### DISCUSSION TITLE
Is there a example for a extension use maven?

### DISCUSSION THREAD

**Discussion by darknesstm** [open-telemetry/opentelemetry-java-instrumentation#11116] [Q&A] [answered]
I don't know much about Gradle. When I directly execute `./gradlew build` as instructed, it prompts an error "Could not resolve all files for configuration ':otel'."

so can give me a example use maven build？

**Comment by laurit** [comment, marked as answer]
> so can give me a example use maven build？

No, we don't have one.
One thing to be careful when using maven for building the extension is that for `InstrumentationModule` classes gradle build has plugins that modify the compiled class files. If you are not using `InstrumentationModule` in your extension then building with maven should work fine, you'll just have to find a maven plugin that generates the SPI file from `@AutoService` annotations or manually generate the. If you wish to use `InstrumentationModule` then you will have to implement `getAdditionalHelperClassNames` method.

> When I directly execute ./gradlew build as instructed, it prompts an error "Could not resolve all files for configuration ':otel'."

where did you execute `./gradlew build`? Anything else in the output?

**Reply by darknesstm** [reply]
Thank you for your reply. I found the reason for the error. It's because I was using sample code for 1.7.x, and the corresponding SNAPSHOT jar no longer exists.

I will also try to learn Gradle. Thanks again.
